src/resources.js

```diff
@@ -475,22 +530,64 @@ export class Resources {
       let minTop = -1;
       for (let i = 0; i < changeHeightRequests.length; i++) {
         let request = changeHeightRequests[i];
+        let resource = request.resource;
         let box = request.resource.getLayoutBox();
         if (box.height == request.newHeight) {
           // Nothing to do.
           continue;
         }
-        if (box.top >= 0) {
-          minTop = minTop == -1 ? box.top : Math.min(minTop, box.top);
+
+        // Check resize rules. It will either resize element immediately, or
+        // wait until scrolling stops or will call the fallback.
+        let resize = false;
+        if (request.force || !this.visible_) {
+          // 1. An immediate execution requested or the document is hidden.
+          resize = true;
+        } else if (activeElement && (activeElement == resource.element ||
+              resource.element.contains(activeElement))) {
+          // 2. Active elements are immediately resized. The assumption is that
+          //    the resize is triggered by the user action or soon after.
+          resize = true;
+        } else if (box.bottom >= viewportRect.bottom - offset) {
+          // 3. Elements under viewport are resized immediately.
+          resize = true;
+        } else if (box.bottom <= viewportRect.top + offset) {
+          // 4. Elements above the viewport can only be resized when scrolling
```

Comment:
I believe I left this case out of the intent-to-implement. Are you adjusting scroll position? I think it would be better if we leave this to a follow up change.